nmt.go

```diff
@@ -170,7 +170,7 @@ func (n *NamespacedMerkleTree) ProveRange(start, end int) (Proof, error) {
 	isMaxNsIgnored := n.treeHasher.IsMaxNamespaceIDIgnored()
 	// TODO: store nodes and re-use the hashes instead recomputing parts of the
 	// tree here
-	if start < 0 || start >= end || end > len(n.leafHashes) {
+	if err := n.validateRange(start, end); err != nil {
 		return NewEmptyRangeProof(isMaxNsIgnored), ErrInvalidRange
```

Comment:
Per the other suggestion, should this be

```suggestion
		return NewEmptyRangeProof(isMaxNsIgnored), err
```

Reply:
Correct, done https://github.com/celestiaorg/nmt/pull/182/commits/d491b529c5e6d7544b126d4879d52a14c785dd69